client-react/src/pages/app/deployment-center/utility/GitHubActionUtility.ts

```diff
@@ -60,6 +60,54 @@ export const updateGitHubActionAppSettingsForPython = async (
   }
 };
 
+export const updateGitHubActionAppSettingsRuntimeStack = async (
+  deploymentCenterData: DeploymentCenterData,
+  resourceId: string,
+  runtimeStack: string,
+  runtimeVersion: string
+) => {
+  const fetchExistingMetadataResponse = await deploymentCenterData.getConfigMetadata(resourceId);
+  if (!fetchExistingMetadataResponse.metadata.success) {
+    LogService.error(LogCategories.deploymentCenter, getLogId('GitHubActionUtility', 'fetchExistingMetadataResponse'), {
+      error: fetchExistingMetadataResponse.metadata.error,
+    });
+
+    return fetchExistingMetadataResponse;
+  }
+
+  const properties = fetchExistingMetadataResponse.data.properties ?? {};
+  properties[DeploymentCenterConstants.metadataCurrentStack] = runtimeStack;
```

Comment:
DIsregard if it is intentional (or inconsequential) to mutate `fetchExistingMetadataResponse.data.properties` in line 79 if it is defined.
```suggestion
  const properties = {
    ...fetchExistingMetadataResponse.data.properties,
    [DeploymentCenterConstants.metadataCurrentStack]: runtimeStack
  };
```